src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -193,7 +193,40 @@ private static void configureSslContext(HttpClientConfig.Builder builder,
                                             ElasticsearchSinkConnectorConfig config) {
     SslFactory kafkaSslFactory = new SslFactory(Mode.CLIENT, null, false);
     kafkaSslFactory.configure(config.sslConfigs());
-    SSLContext sslContext = kafkaSslFactory.sslContext();
+
+    SSLContext sslContext;
+    try {
+      // try AK <= 2.2 first
+      sslContext =
+          (SSLContext) SslFactory.class.getDeclaredMethod("sslContext").invoke(kafkaSslFactory);
+      log.trace("Using AK 2.2< SslFactory.");
+    } catch (Exception e) {
+      // must be running AK 2.3+
+      log.trace("Could not find SslFactory::sslContext. Trying AK 2.3+ APIs.");
+
+      Object ss;
+      try {
+        // try AK <= 2.6 second
+        ss = SslFactory.class.getDeclaredMethod("sslEngineBuilder").invoke(kafkaSslFactory);
+        log.trace("Using AK 2.6< SslFactory.");
+
+      } catch (Exception ex) {
+        // must be running AK 2.6+
+        log.trace("AK 2.6< SslFactory not found. Trying AK 2.6+.");
```

Comment:
we should word such errors so that users not familiar with the internals can understand what is going on.

```suggestion
        log.trace("SslFactory not found. Switching to SslEngine libs from AK 2.6 and beyond.");
```

might want to take another pass at the other trace statements.